File: articles/why-moral-psychology-matters-today/fortran/relevance_score.f90
! Synthetic relevance score example for "Why Moral Psychology Matters Today".
! Educational scaffold only.
!
! Compile:
!   gfortran fortran/relevance_score.f90 -o outputs/tables/relevance_score_fortran
! Run:
!   ./outputs/tables/relevance_score_fortran

program relevance_score
  implicit none

  real :: political_polarization
  real :: organizational_risk
  real :: developmental_learning
  real :: cross_cultural_variation
  real :: technology_amplification
  real :: moral_injury_burden
  real :: score

  political_polarization = 1.2
  organizational_risk = 0.8
  developmental_learning = 0.4
  cross_cultural_variation = 0.5
  technology_amplification = 1.0
  moral_injury_burden = 0.9

  score = 0.25 * political_polarization + &
          0.20 * organizational_risk + &
          0.15 * developmental_learning + &
          0.15 * cross_cultural_variation + &
          0.15 * technology_amplification + &
          0.20 * moral_injury_burden

  print *, "Synthetic moral psychology relevance score:", score
end program relevance_score

File: articles/why-moral-psychology-matters-today/rust/relevance_score.rs
// Synthetic relevance score example for "Why Moral Psychology Matters Today".
// Educational scaffold only.
//
// Compile:
//   rustc rust/relevance_score.rs -o outputs/tables/relevance_score_rust
// Run:
//   ./outputs/tables/relevance_score_rust

fn main() {
    let variables = [
        ("political_polarization", 1.2_f64, 0.25_f64),
        ("organizational_risk", 0.8, 0.20),
        ("developmental_moral_learning", 0.4, 0.15),
        ("cross_cultural_variation", 0.5, 0.15),
        ("technology_amplification", 1.0, 0.15),
        ("moral_injury_burden", 0.9, 0.20),
    ];

    let score: f64 = variables
        .iter()
        .map(|(_, value, weight)| value * weight)
        .sum();

    println!("Synthetic moral psychology relevance score: {:.3}", score);
}
